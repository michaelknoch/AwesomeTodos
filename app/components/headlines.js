import React, { PropTypes } from 'react';
import { StyleSheet, Text } from 'react-native';

const propTypes = {
    children: PropTypes.string.isRequired,
    style: PropTypes.object,
};

const styles = StyleSheet.create({
    headline: { color: 'white' },
    h1: {
        fontSize: 70,
    },
    h2: {
        fontSize: 50,
    },
    h3: {
        fontSize: 30,
    },
});

export const H1 = ({ children, style }) => (
    <Text style={[styles.headline, styles.h1, style]}>{children}</Text>
);

export const H2 = ({ children, style }) => (
    <Text style={[styles.headline, styles.h2, style]}>{children}</Text>
);

export const H3 = ({ children, style }) => (
    <Text style={[styles.headline, styles.h3, style]}>{children}</Text>
);

H1.propTypes = propTypes;
H2.propTypes = propTypes;
H3.propTypes = propTypes;
