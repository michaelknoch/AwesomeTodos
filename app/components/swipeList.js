import React, { Component } from 'react';
import { View, ListView, StyleSheet, Text } from 'react-native';
import Swipeout from 'react-native-swipeout';

const styles = StyleSheet.create({
    wrapper: {
        flex: 1,
        flexDirection: 'column',
    },
    row: {
        height: 50,
        justifyContent: 'center',
        padding: 10,
        backgroundColor: 'rgb(50,50,50)',
    },
});

class SwipeList extends Component {

    render() {
        const { removeTodo } = this.props;
        const ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });
        this.state = {
            dataSource: ds.cloneWithRows(this.props.todos),
        };

        const renderRow = (rowData) => {
            const swipeoutBtns = [
                {
                    text: 'X',
                    backgroundColor: '#e74c3c',
                    onPress: () => {
                        this.props.removeTodo(rowData.id);
                    },
                },
            ];

            return (
                <Swipeout backgroundColor={'rgb(50,50,50)'} autoClose left={swipeoutBtns}>
                    <View style={styles.row}>
                        <Text style={{ color: 'white' }}>{rowData.text}</Text>
                    </View>
                </Swipeout>
            );
        }

        return (
            <View style={{ flex: 1 }}>
                <ListView
                    styles={styles.wrapper}
                    dataSource={this.state.dataSource}
                    renderRow={renderRow}
                />
            </View>
        );
    }
}

export default SwipeList;
