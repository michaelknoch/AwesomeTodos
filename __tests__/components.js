import React from 'react';
import renderer from 'react-test-renderer';

import SwipeList from '../app/components/swipeList';
import { H1, H2, H3 } from '../app/components/headlines';

const todos = [
    { id: 0, text: 'initial' },
    { id: 1, text: 'todos' },
];

describe('swipeList component', () => {
    it('should render correctly', () => {
        const tree = renderer.create(
            <SwipeList todos={todos} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('headlines', () => {
    it('should render H1 correctly', () => {
        const tree = renderer.create(
            <H1>headline 1</H1>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render H2 correctly', () => {
        const tree = renderer.create(
            <H2>headline 2</H2>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render H3 correctly', () => {
        const tree = renderer.create(
            <H3>headline 3</H3>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
