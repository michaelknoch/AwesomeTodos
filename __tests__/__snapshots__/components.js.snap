exports[`swipeList component should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <ScrollView
    dataSource={
      ListViewDataSource {
        "items": 2,
      }
    }
    initialListSize={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    removeClippedSubviews={true}
    renderRow={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}
    styles={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#dbddde",
            "overflow": "hidden",
          },
          undefined,
        ]
      }>
      <View
        onLayout={[Function]}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Array [
            Object {},
            Object {
              "left": 0,
            },
          ]
        }>
        <View
          style={
            Object {
              "backgroundColor": "rgb(50,50,50)",
              "height": 50,
              "justifyContent": "center",
              "padding": 10,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
              }
            }>
            initial
          </Text>
        </View>
      </View>
      <View />
      <View />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#dbddde",
            "overflow": "hidden",
          },
          undefined,
        ]
      }>
      <View
        onLayout={[Function]}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Array [
            Object {},
            Object {
              "left": 0,
            },
          ]
        }>
        <View
          style={
            Object {
              "backgroundColor": "rgb(50,50,50)",
              "height": 50,
              "justifyContent": "center",
              "padding": 10,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
              }
            }>
            todos
          </Text>
        </View>
      </View>
      <View />
      <View />
    </View>
  </ScrollView>
</View>
`;
